supervisor/archives_api.go

```diff
@@ -26,6 +27,14 @@ func (self ArchiveAPI) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		if status != "" {
 			desiredStatus = []string{status}
 		}
+		limit := paramValue(req, "limit", "")
+		if limit != "" {
+			limint, err := strconv.Atoi(limit)
+			if err != nil || limint <= 0 {
+				bailWithError(w, ClientErrorf("invalid limit supplied"))
+				return
+			}
+		}
```

Comment:
We should refactor this (and the other occurrence from the other day) into a `validateLimit(string) error` function that will validate the passed strings as a non-negative value and return the `ClientErrorf(...)` call.